keras_cv/layers/object_detection3d/utils.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+
+def HasRank(tensor, expected_rank):
```

Comment:
Let's move these utils to keras_cv/utils/tensor_utils.py and use lowercase_underscore naming convention (to match other utils).

It seems like these methods aren't 3D OD specific, so a more generic codepath for them probably makes more sense

Reply:
Is it possible to make this private utils for now, before we consider how to generalize it and move to keras_cv/utils?

Reply:
Add '_' prefix to indicate it is internal and add comments to docstring.